treasury-fees-monitor/src/agent.spec.js

```diff
@@ -0,0 +1,357 @@
+const mockContract = {};
+const mockReadStream = jest.fn();
+
+jest.mock('forta-agent', () => ({
+  ...jest.requireActual('forta-agent'),
+  getEthersProvider: jest.fn(),
+  ethers: {
+    ...jest.requireActual('ethers'),
+    Contract: jest.fn().mockReturnValue(mockContract),
+  },
+}));
+
+jest.mock('fs', () => ({
+  ...jest.requireActual('fs'),
+  createReadStream: mockReadStream,
+}));
+
+const {
+  Finding, FindingType, FindingSeverity, ethers, TransactionEvent,
+} = require('forta-agent');
+
+const { provideHandleTransaction, provideInitialize } = require('./agent');
+const {
+  getObjectsFromAbi,
+  createMockEventLogs,
+} = require('./test-utils');
+const { getAbi, calculateStatistics, parseCsvAndCompute } = require('./utils');
+const config = require('../bot-config.json');
+
+// setup mock fs streaming object
+const mockAssetToken = '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2';
+const mockStream = {
+  pipe: jest.fn().mockReturnThis(),
+  // eslint-disable-next-line func-names
+  on: jest.fn().mockImplementation(function (event, handler) {
+    if (event === 'data') {
+      handler({
+        asset: mockAssetToken,
+        premium: '1000000000000000000',
+      });
+    } else {
+      handler();
+    }
+
+    return this;
+  }),
+};
+
+// utility function specific for this test module
+// we are intentionally not using the Forta SDK function due to issues with
+// jest mocking the module and interfering with default function values
+function createTransactionEvent(txObject) {
+  const txEvent = new TransactionEvent(
+    null,
+    null,
+    null,
+    [],
+    {},
+    null,
+    txObject.logs,
+    null,
+  );
+  return txEvent;
+}
+
+// check the configuration file to verify the values
+describe('check bot configuration file', () => {
+  it('protocolName key required', () => {
+    const { protocolName } = config;
+    expect(typeof (protocolName)).toBe('string');
+    expect(protocolName).not.toBe('');
+  });
+
+  it('protocolAbbreviation key required', () => {
+    const { protocolAbbreviation } = config;
+    expect(typeof (protocolAbbreviation)).toBe('string');
+    expect(protocolAbbreviation).not.toBe('');
+  });
+
+  it('developerAbbreviation key required', () => {
+    const { developerAbbreviation } = config;
+    expect(typeof (developerAbbreviation)).toBe('string');
+    expect(developerAbbreviation).not.toBe('');
+  });
+
+  it('contracts key required', () => {
+    const { contracts } = config;
+    expect(typeof (contracts)).toBe('object');
+    expect(contracts).not.toBe({});
```

Comment:
We probably have this line elsewhere, but I don't think it will do what we want it to do, much like the other times we have attempted to check for an empty Object.

It probably needs to be more like:
```suggestion
    expect(Object.keys(contracts).length).not.toBe(0);
```